litestar/middleware/exceptions/middleware.py

```diff
@@ -147,17 +148,30 @@ class ExceptionHandlerMiddleware:
     This used in multiple layers of Litestar.
     """
 
-    def __init__(self, app: ASGIApp, debug: bool, exception_handlers: ExceptionHandlersMap) -> None:
+    def __init__(self, app: ASGIApp, debug: bool | None, exception_handlers: ExceptionHandlersMap) -> None:
         """Initialize ``ExceptionHandlerMiddleware``.
 
         Args:
             app: The ``next`` ASGI app to call.
-            debug: Whether ``debug`` mode is enabled
+            debug: Whether ``debug`` mode is enabled. Deprecated. Debug mode will eb inferred from the request scope
             exception_handlers: A dictionary mapping status codes and/or exception types to handler functions.
         """
         self.app = app
         self.exception_handlers = exception_handlers
         self.debug = debug
+        if debug is not None:
+            warn_deprecation(
+                "2.0.0rc2",
```

Comment:
im not sure we will be getting there, maybe just do 2.0.0

Reply:
Well, it would be the next major version then, so 3.
But I agree, 2.0.0rc is not a good target. I'll change this.